src/component/Reduxpage.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import Header from "./redux/header";
+import List from "./redux/list";
+
+function Homepage() {
+  const id = useSelector((state) => state.id);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (id === 0) {
+      fetch(
+        "https://raw.githubusercontent.com/JiaAnTW/json_storage/master/react-tutorial/todolist.json",
+        { method: "GET" }
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          console.log(result.data.length);
+          dispatch({ type: "Value", value: result.data.length + 1 });
+          dispatch({ type: "InitTask", allTask: result.data });
+        });
+    }
+  }, [id, dispatch]);
```

Comment:
Please use `redux-thunk` to call API instead of using useEffect.

Reply:
![image](https://user-images.githubusercontent.com/81241494/142750377-f5ad6591-a1dc-4988-a0da-7b0bc9f34661.png)

Reply:
@Kyl23 That's native Redux with React hook. Please check this ref:
https://ithelp.ithome.com.tw/articles/10253094